lib/cli-engine.js

```diff
@@ -110,6 +110,51 @@ function loadPlugins(pluginNames) {
 }
 
 /**
+ * It will calculate the error and warning count for collection of massages per file
+ * @param {Object[]} messages - Collection of messages
+ * @returns {Object} Contains the stats
+ * @private
+ */
+function statsPerFile(messages) {
+    var errorCount = 0,
+        warningCount = 0;
+
+    messages.forEach(function(message) {
+        if (message.fatal || message.severity === 2) {
+            errorCount++;
+        } else {
+            warningCount++;
+        }
+    });
+
+    return {
+        errorCount: errorCount,
+        warningCount: warningCount
+    };
+}
+
+/**
+ * It will calculate the error and warning count for collection of results from all files
+ * @param {Object[]} results - Collection of messages from all the files
+ * @returns {Object} Contains the stats
+ * @private
+ */
+function statsPerRun(results) {
```

Comment:
Did you consider using `reduce` instead of `forEach` here and above?

```js
return results.reduce(function(stats, result) {
    stats.errorCount += result.errorCount;
    stats.warningCount += result.warningCount;
    return stats;
}, {
    errorCount: 0,
    warningCount: 0
});
```